docs/runtime_image.md

```diff
@@ -0,0 +1,83 @@
+| `/tmp/app.war:.`        | the same as above |
+| `/tmp/app.war:app/dist` | `/tmp/app.war` file will be copied from a builder image into into the `app/dist` subdirectory of the `WORKDIR` of a runtime image |
+| `/opt/data`             | `/opt/data` directory will be copied from a builder image into the `WORKDIR` of a runtime image |
+
+Keep in mind that the `source` must be an absolute path. The `destination` must be a relative path to directory, it must exist and path must not start from `..` 
+
+During copying the artifacts, S2I will modify their permissions. All directories and files with executable bit will be uploaded with `0755` mode. Other files will have `0644` mode.
+
+### `assemble-runtime` script requirements
+
+`assemble-runtime` script can be written in any programming language. S2I searches the following locations for this script in the following order:
+
+1. A script found at the `--scripts-url` URL
+1. A script found in the application source `.s2i/bin` directory
+1. A script found at the default image URL (`io.openshift.s2i.scripts-url` label)
+
+`assemble-runtime` script is always gets executed under `USER` from a runtime image.
+
+### Runtime image requirements
+
+In most of the cases you can use any images as a runtime image. Only if you are using `--allowed-uids` option then image should have numeric `USER` and this user should be within the range of allowed users.
+
+To simplify build workflow and to provide some reasonable defaults, author of the runtime image can use the following techniques:
+
+* `run` and `assemble-runtime` scripts can be placed inside of the runtime image. Scripts from the image will be used as a fallback when user didn't have any in the `.s2i/bin` directory of his repository and when he didn't specify alternative location with `--scripts-url` option. The location of the scripts is defined by the value of the `io.openshift.s2i.scripts-url` label that should be presented on the image. For example, you can set it to `image:///usr/libexec/s2i`
+* default mapping for the files can be specified by adding `io.openshift.s2i.assemble-input-files` label to a runtime image. This mapping will be used as a fallback when user didn't provide it explicitly with `--runtime-artifact` option.
+
+To specify mapping for multiple files separate them by a semicolon. For example: `/tmp/app.war:app;/opt/data`
+
+### Build and runtime environments
+
+Builder and runtime containers have the same environment. In other words `assemble` and `assemble-runtime` scripts are able to use environment variables defined with `--env` and `--environment-file` options along with the values from `.s2i/environment` file in the repo.
+
+### Extended build and incremental build
+
+In the current implementation it's impossible to do an exteneded incremental build. This combination is invalid and build will fail.
```

Comment:
s/it's impossible/it is not possible/
s/exteneded/extended/
s/and build will fail/and the build will fail/